src/engine/strat_engine/crypt/shared.rs

```diff
@@ -838,20 +484,44 @@ pub fn activate(
             )));
         }
     }
-    log_on_failure!(
+    if let Some(p) = passphrase {
```

Comment:
I'm proposing a restructuring like the following:

```
let key_slot = ...
if let Some(p) = passphrase {
  log_on_failure!(device.activate_handle()...
} else {
  device.token_handle()...
}
```
i.e., hoisting the ```key_slot``` assignment out of the enclosing if. The reason for that is to avoid somewhat ugly error messages like:

```$ sudo stratis pool start
--name spool1 --unlock-method=clevis
Execution failed:
stratisd failed to perform the operation that you requested. It
returned the following information via the D-Bus: ERROR: Cryptsetup
error: IO error occurred: No such file or directory (os error 2).
$ sudo stratis pool start
--name spool2 --unlock-method=clevis
Execution failed:
stratisd failed to perform the operation that you requested. It
returned the following information via the D-Bus: ERROR: An attempt to
set up pool with UUID 9b8eb004-c479-478e-9406-13c8f786add4 from the
assembled devices failed; Cryptsetup error: IO error occurred: No such
file or directory (os error 2).
```

which you can get when you have encrypted with a key but you specify unlock method as clevis.

Reply:
I'm not confident this is the best way to handle it only because keyslots are entirely bound to the passphrase implementation. Would augmenting the error handling for activating by token to detect errno 2 and return a better error message be sufficient? I did some digging into the cryptsetup source code and it looks like this will only be returned on failure to fetch that token/keyslot.

Reply:
Yes. That would also be fine.

Reply:
I tested my change and it works as expected.